### ISSUE TITLE
Bazel: Allowing .zip archive URLs in addition to .tar.gz

### ISSUE THREAD

**Issue by cpsauer** [renovatebot/renovate#7851]
**What would you like Renovate to be able to do?**

For upgrading GitHub http_archives in a Bazel WORKSPACE, it'd be great if Renovate would accept .zip URLs in addition to those with a .tar.gz suffix.

Motivation: While it's easy enough to get Renovate to pick up dependencies by manually changing all the URLs over to their .tar.gz equivalents at the beginning, I'm concerned about collaborators adding dependencies with .zip URLs later--especially since .zip is the default/top archive link on GitHub. In particular, I'm concerned about having those .zip URLs silently fail to be updated. 

Currently, URLs ending in .zip seem to be silently ignored as not needing an update, with indications that they they'd failed only found in the Renovate log dashboard. The currentValue gets listed as "[value].zip" and then skipped, with skipReason "unsupported-value". If I'm missing a way to get notified about unrecognized deps and skips, lmk (that would be great, too!). But it seems like this instance probably has an easy fix (see below), and feels a bit like a bug/oversight, again unless I'm missing something.

**Did you already have any implementation ideas?**

I think this is as easy as splicing out the .zip extensions in this line in extract.ts, in addition to .tar.gz:  https://github.com/renovatebot/renovate/blob/c63ffa5d6232da8e68186389561531ac8f3c5943/lib/manager/bazel/extract.ts#L34
It doesn't matter to me if .zip URLs were replaced with .tar.gz on update, as I think the simple change above would do. My needs are about input tolerance, though output extension consistency might matter to others.
Also, assuming the fix is that easy, I think we should consider handling .tar.bz2 and maybe .tar at the same time. That way we'd have all the GitHub/GitLab supported options that I've seen. If we wanted to be more complete to what Bazel's [http_archive supports](https://docs.bazel.build/versions/master/repo/http.html#http_archive), we could add .tgz and .tar.xz also. Happy to help if you'd like.

Thanks for your consideration!
Chris

**Comment by rarkins** [comment]
No objection to adding this, and a PR would be welcome if you have the time. 

IIRC our file name logic was quite brittle and assumes the file structure doesn't change. There's probably room to improve that by explicitly checking the release file list, although it would be a separate improvement to this one you raise.

**Comment by cpsauer** [comment]
@rarkins, sorry for the delay, but I finally got a chance to take a crack at it. PR #7929 

Tried to do the fullest version of the above: Accepting all the archive types mentioned (for future proofing), preserving extensions on update, and, to solve the extension hardcoding updating, unifying the archive/tag/version split mentioned in #7852.

It's my first Renovate (and TypeScript) commit, so I'd really appreciate your coaching and patience if I've missed something :)

Thanks!
Chris
